Sources/KSCrashRecording/Monitors/KSCrashMonitor_Memory.h

```diff
@@ -0,0 +1,93 @@
+//
+//  KSCrashMonitor_Memory.h
+//
+//  Created by Alexander Cohen on 2024-05-20.
+//
+//  Copyright (c) 2024 Alexander Cohen. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall remain in place
+// in this source code.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+
+/* Monitor memory and records data for OOMs.
+ */
+
+
+#ifndef KSCrashMonitor_Memory_h
+#define KSCrashMonitor_Memory_h
+
+#include "KSCrashMonitor.h"
+#include "KSCrashAppTransitionState.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ App Memory
+ */
+typedef struct KSCrash_Memory {
+    
+    /** timestamp in microseconds */
+    int64_t timestamp;
```

Comment:
What do you think about adding the `int8_t version` as a first field here as an extra safety measure? And the size of the structure (yet it's unlikely to change).
And maybe even with a few ["magic bytes"](https://en.wikipedia.org/wiki/Magic_number_(programming)#In_files) as `int32_t magic`.

Reply:
great Idea. :)